src/IK.cpp

```diff
@@ -0,0 +1,103 @@
+// This file is part of RBDyn.
+//
+// RBDyn is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// RBDyn is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with RBDyn.  If not, see <http://www.gnu.org/licenses/>.
+
+// associated header
+#include "IK.h"
+
+// includes
+// RBDyn
+#include "FK.h"
+#include "FV.h"
+#include "MultiBody.h"
+#include "MultiBodyConfig.h"
+
+//SpaceVecAlg
+#include <SpaceVecAlg/SpaceVecAlg>
+
+namespace rbd
+{
+
+static const int MAX_IK_ITERATIONS = 50;
+static const double LAMBDA = 0.9;
+static const double IK_THRESHOLD = 1e-8;
+static const double ALMOST_ZERO = 1e-8;
```

Comment:
This should be in an anonymous namespace to avoid future name conflicts or declare them `static constexpr` in `IK.h` and use as default arguments for the algorithm construction.